rfcs/rfc-bn-messaging-queue.md

```diff
@@ -0,0 +1,285 @@
+    <CCRAHeader>
+      <userApplication>BI</userApplication>
+      <userRole>01</userRole>
+    </CCRAHeader>
+    <partnerUserID>BCSBNHUB</partnerUserID>
+  </header>
+  <body>
+    <legalName>Patrick Wei</legalName>
+    <businessTypeCode>01</businessTypeCode>
+    <postalZIPCode>V8W3E6</postalZIPCode>
+  </body>
+</SBNClientBasicInformationSearchRequest>
+```
+
+Response Sample
+```xml
+<?xml version="1.0"?>
+<SBNClientBasicInformationSearchResponse>
+	<header>
+		<requestMode>S</requestMode>
+		<documentSubType>000</documentSubType>
+		<senderID>CCRABN</senderID>
+		<receiverID>BCSBNHUB</receiverID>
+		<transactionID>BNTVO6C1B8J</transactionID>
+		<transactionDateTime>2022-04-07-15.31.55.159000</transactionDateTime>
+		<CCRAHeader>
+			<userApplication>BI</userApplication>
+			<userRole>01</userRole>
+		</CCRAHeader>
+	</header>
+
+	<body>
+		<clientBasicInformationSearchInformationMessage>
+			<messageNumber>70377</messageNumber>
+			<messageEnglishText>Exact Match</messageEnglishText>
+			<messageFrenchText>Concordance exacte</messageFrenchText>
+		</clientBasicInformationSearchInformationMessage>
+		<clientBasicInformationSearchResult>
+			<businessRegistrationNumber>993775204</businessRegistrationNumber>
+			<legalName>Patrick Wei</legalName>
+			<operatingName>ROY&#39;S SOLE PROP</operatingName>
+			<businessTypeCode>01</businessTypeCode>
+			<corporationCertification>
+				<incorporationDate />
+				<incorporationCertificateIdentifier />
+				<jurisdictionCode />
+				<jurisdictionProvinceCode />
+				<jurisdictionCountryCode />
+				<jurisdictionStateCode />
+			</corporationCertification>
+			<businessAddress>
+				<foreignLegacy>
+					<addressDetailLine1>940 BLANSHARD ST</addressDetailLine1>
+					<addressDetailLine2 />
+				</foreignLegacy>
+				<municipality>VICTORIA</municipality>
+				<provinceStateCode>BC</provinceStateCode>
+				<postalCode>V8W3E6</postalCode>
+				<countryCode>CA</countryCode>
+				<returnMailIndicator>0</returnMailIndicator>
+			</businessAddress>
+			<mailingAddress>
+				<foreignLegacy>
+					<addressDetailLine1>940 BLANSHARD ST</addressDetailLine1>
+					<addressDetailLine2 />
+				</foreignLegacy>
+				<municipality>VICTORIA</municipality>
+				<provinceStateCode>BC</provinceStateCode>
+				<postalCode>V8W3E6</postalCode>
+				<countryCode>CA</countryCode>
+				<effectiveDate>2022-08-17</effectiveDate>
+				<returnMailIndicator>0</returnMailIndicator>
+				<careOfLine />
+			</mailingAddress>
+			<clientEvents>
+				<eventCode>03</eventCode>
+				<effectiveDate>2022-08-17</effectiveDate>
+				<expiryDate>9999-12-31</expiryDate>
+			</clientEvents>
+			<relatedBusinessRegistrationNumber />
+			<owners>
+				<ownerIndividualName>
+					<lastName>Wei</lastName>
+					<givenName>Patrick</givenName>
+				</ownerIndividualName>
+				<ownerBN />
+			</owners>
+		</clientBasicInformationSearchResult>
+	</body>
+</SBNClientBasicInformationSearchResponse>
+```
+
+# Approach
```

Comment:
I'm a little confused.  Are 1 and 2 options?  Or is the proposed solution to implement both a queue and a job?

Reply:
Yes. It's options. I initially thought the queue would be enough to handle these calls. But when I realized we need a buffer time (of at least 30 seconds) between 1st and 2nd call I thought of the 2nd option

Reply:
I think the 1st option is nice in that we might get the BN number asap.  But I feel like the 2nd option is the safer and probably more reliable option.

Reply:
Yeah! I have the same feeling

Reply:
we usually have both in place.
the job could open a q msg to poll if you wanted to keep the same mechanism and a count of tries
-or-
as is done in our other queues, raise it as an error which puts it back on the queue for a retry. keep a count of the retries and bail with an ops level error message to Sentry for manual intervention.